### PR TITLE
Support sap sms 365 digital interconnect

### PR DESCRIPTION
**Pull request by thiperson**
Hello,

Please consider this PR to add support for SAP digital interconnect API (which is now owned by Sinch)

Here is a link to documentation : https://community.digitalinterconnect.com/t5/Enterprise-Services-Knowledgebase/SMS-365-enterprise-service-HTTP-Interface-Specification/ta-p/745

Technical specification is here : https://community.digitalinterconnect.com/t5/Enterprise-Services-Knowledgebase/SMS-365-enterprise-service-HTTP-Interface-Specification/ta-p/745?attachment-id=1055

Thanks

### CODE REVIEW

**Comment by marcelcorso** [comment]
hey @thiperson! CI was failing because we were using a very old go version on it. 
I updated it on master. Please merge it into your branch and update the PR so the CI can run again?

**Comment by thiperson** [comment]
Hi @marcelcorso, I just merged and pushed. CI seems OK.
I'm open to any suggestion as it is my first PR on Github, and first time playing with Go as well.

File: provider/sap/sap.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	URL      string `yaml:"url"`
+	AuthHash string `yaml:"auth_hash"`
+}
+
+// Sap contains the necessary values for the Sap provider
+type Sap struct {
+	Config
+}
+
+var sapHTTPClient = &http.Client{Timeout: time.Second * 20}
```
Comment (marcelcorso) [inline provider/sap/sap.go:22]: `sapHTTPClient` doesn't need to be a package variable. You can instantiate it on the `NewSap` function and store in the Sap struct. 
Then on the Send `func` you can use it like `c.client.Do(request)`

**Comment by marcelcorso** [comment]
looking good. I just added a small comment 👍

File: provider/sap/sap.go (hunk 1 above)
Comment (thiperson) [reply provider/sap/sap.go]: Done !

**Pull request merged**